nixos/modules/services/networking/nat.nix

```diff
@@ -81,7 +81,7 @@ let
 
           iptables -w -t nat -A nixos-nat-post \
             -d ${destinationIP} -p ${fwd.proto} \
-            --dport ${destinationPorts} \
+            --dport ${builtins.replaceStrings ["-"] [":"] destinationPorts} \
```

Comment:
I think it would be better to do it at line 68:  there is PR (#78315) that when merged will use `destinationPorts` several times. Try something like

`destinationPorts = ... else buitins.replaceStrings ["-"] [":"] (elemAt m 1)`